script.js

```diff
@@ -0,0 +1,309 @@
+
+
+
+let list = document.getElementById("list");
+let listToDo = JSON.parse(localStorage.getItem("list"));
+
+
+
+if (listToDo?.length === 0 || !listToDo) {
+  listToDo = [];
+}
+
+function updatePage() {
+  list.innerHTML = null;
+  for (let i = 0; i < listToDo.length; i++) {
+    createOnPg(listToDo[i].taskName, listToDo[i].date, listToDo[i].tag);
+  }
+  refreshLocalStorage(listToDo);
+}
+
+taskInputForm.addEventListener("submit", (e) => {
+  e.preventDefault();
+});
+
+taskNameInput.addEventListener("focus", () => {
+  dateInput.setAttribute("id", "date-appear");
+  dateInput.disabled = false;
+  setTimeout(() => { 
+    tagInput.setAttribute("id", "tag-appear");
+    tagInput.disabled = false; 
+  }, 300);
+});
+
+removeOddBtn.addEventListener("click", () => {
+  for (let i = 0; i < listToDo.length; i++) {
+    if (i % 2 === 0) {
+      listToDo.splice(i, 1);
+    }
+  }
+  updatePage();
+})
+
+removeDupBtn.addEventListener("click", () => {
+  const tempSet = new Set();
+  const tempArray = [];
+  for (let i = 0; i < listToDo.length; i++) {
+    if (!tempSet.has(listToDo[i].taskName)) {
+      tempArray.push(listToDo[i]);
+    }
+    tempSet.add(listToDo[i].taskName);
+  }
+  listToDo = tempArray;
+  updatePage();
+});
+
+removeRandomBtn.addEventListener("click", () => {
+  const random = Math.round(Math.random() * listToDo.length - 1);
+  listToDo.splice(random, 1);
+  updatePage();
+});
+
+removeLastBtn.addEventListener("click", () => {
+  listToDo.pop();
+  updatePage();
+});
+
+removeFirstBtn.addEventListener("click", () => {
+  listToDo.shift();
+  updatePage();
+});
+
+addRandomBtn.addEventListener("click", () => {
+  const randomNum = Math.round(Math.random() * listToDo.length - 1);
+  const newItem = {
+    taskName: taskNameInput.value,
+    date: dateInput.value,
+    tag: tagInput.value
+  }
+  listToDo.splice(randomNum, 0, newItem);
+  updatePage();
+});
+
+addFirstBtn.addEventListener("click", () => {
+  const newItem = {
+    taskName: taskNameInput.value,
+    date: dateInput.value,
+    tag: tagInput.value
+  }
+  listToDo.unshift(newItem);
+  updatePage();
+});
+
+randomizeBtn.addEventListener("click", () => {
+  listToDo.sort(() => Math.random() - 0.5);
+  updatePage();
+});
+
+alphaBtnOrderBtn.addEventListener("click", () => {
+  listToDo.sort((a, b) => {
+    const taskA = a.taskName.toUpperCase();
+    const taskB = b.taskName.toUpperCase();
+    if (taskA > taskB) {
+      return 1;
+    }
+    if (taskA < taskB) {
+      return -1;
+    }
+    return 0;
+  });
+  updatePage();
+});
+
+alphaRvrsOrderBtn.addEventListener("click", () => {
+  listToDo.sort((a, b) => {
+    const taskA = a.taskName.toUpperCase();
+    const taskB = b.taskName.toUpperCase();
+    if (taskA > taskB) {
+      return -1;
+    }
+    if (taskA < taskB) {
+      return 1;
+    }
+    return 0;
+  });
+
+  updatePage();
+});
+
+window.addEventListener("load", () => {
+  for (let i = 0; i < listToDo.length; i++) {
```

Comment:
tenta usar as funções dos arrays quando podes, é mais legível: listToDo.forEach(({ taskName, date, tag }) => createOnPg(taskName, date, tag));

Reply:
tens outros que podem ficar assim também

Reply:
ou este é aquele caso que não dá para usar .forEach? 
nesse caso acho que poderias fazer `for (let item of listToDo) {...}` ou `Array.from(listToDo).forEach()`

Reply:
Ok ok compreendi